lib/aws-xray-sdk/plugins/ecs.rb

```diff
@@ -10,13 +10,100 @@ module ECS
 
       ORIGIN = 'AWS::ECS::Container'.freeze
 
+      METADATA_BASE_URL = 'http://169.254.169.254/latest'.freeze
+
       def self.aws
-        @@aws ||= begin
```

Comment:
The new implementation have been mostly inspired by: 
https://github.com/aws/aws-xray-sdk-ruby/blob/master/lib/aws-xray-sdk/plugins/ec2.rb

As well as by the java counter part implementation:
https://github.com/aws/aws-xray-sdk-java/blob/398ec0542964b4d9b38c901385eda4d98da07cd4/aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/plugins/ECSMetadataFetcher.java
https://github.com/aws/aws-xray-sdk-java/blob/398ec0542964b4d9b38c901385eda4d98da07cd4/aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/plugins/ECSPlugin.java